core/src/main/scala/cats/evidence/As.scala

```diff
@@ -0,0 +1,201 @@
+package cats
+package evidence
+
+import arrow.Category
+
+/**
+ * As substitutability: A better `<:<`
+ *
+ * This class exists to aid in the capture proof of subtyping
+ * relationships, which can be applied in other context to widen other
+ * type
+ *
+ * `A As B` holds whenever `A` could be used in any negative context
+ * that expects a `B`.  (e.g. if you could pass an `A` into any
+ * function that expects a `B` as input.)
+ *
+ * This code was ported directly from scalaz to cats using this version from scalaz:
+ * https://github.com/scalaz/scalaz/blob/a89b6d63/core/src/main/scala/scalaz/As.scala
+ *
+ *  The original contribution to scalaz came from Jason Zaugg
+ */
+sealed abstract class As[-A, +B] extends Serializable {
+  /**
+   * Use this subtyping relationship to replace B with a value of type
+   * A in a contravariant context.  This would commonly be the input
+   * to a function, such as F in: `type F[-B] = B => String`. In this
+   * case, we could use A As B to turn an F[B] Into F[A].
+   */
+  def substitute[F[-_]](p: F[B]): F[A]
+
+  @inline final def andThen[C](that: (B As C)): (A As C) = As.compose(that, this)
+
+  @inline final def compose[C](that: (C As A)): (C As B) = As.compose(this, that)
+
+  @inline final def coerce(a: A): B = As.witness(this)(a)
+}
+
+sealed abstract class AsInstances {
+  import As._
+
+  /*
+   * Subtyping forms a category
+   */
+  implicit val liskov: Category[As] = new Category[As] {
+    def id[A]: (A As A) = refl[A]
+
+    def compose[A, B, C](bc: B As C, ab: A As B): (A As C) = bc compose ab
+  }
+}
+
+object As extends AsInstances {
+  /**
+   * In truth, "all values of `A Is B` are `refl`". `reflAny` is that
+   * single value.
+   */
+  private[this] val reflAny = new (Any As Any) {
+    def substitute[F[-_]](fa: F[Any]) = fa
+  }
+  /**
+   * Subtyping is reflexive
+   */
+  implicit def refl[A]: (A As A) =
+    reflAny.asInstanceOf[A As A]
+
+  /**
+   * We can witness the relationship by using it to make a substitution *
+   */
+  implicit def witness[A, B](lt: A <~< B): A => B =
+    lt.substitute[-? => B](identity)
+
+  /**
+   * Subtyping is transitive
+   */
+  def compose[A, B, C](f: B <~< C, g: A <~< B): A <~< C =
+    g.substitute[λ[`-α` => α <~< C]](f)
+
+  /**
+   * Lift Scala's subtyping relationship
+   */
+  @inline def unsafeFromPredef[A, B >: A]: (A As B) =
+    reflAny.asInstanceOf[A As B]
```

Comment:
So, start with refl, then do to covariance you have `A As A <: A As B` so it can be returned. So I think returning `refl[A]` here will compile.